frontend/script.js

```diff
@@ -50,104 +36,213 @@ const init = () => {
         
         newArr.forEach(element => {
             
-            
-            
-            li.innerHTML = element
-
-            otherBtn.append(editBtn)
-            otherBtn.append(deleteBtn)
-
-            displayTodo.appendChild(checkbox)
+            li.innerHTML = `<input type='checkbox' data-check='check'> ${element} <div class = 'otherbtn'><button data-edit='edit' class='editBtn'>Edit</button> <button data-delete='delete'>Delete</button></div>`
+                
             displayTodo.appendChild(li)
-            displayTodo.appendChild(otherBtn)
 
         });
 
-        props.classList.remove('hidden')
-        postData()
-        
-       
+        hideDivs()
+        // postData()
+         
     }
     
     input.value =  ""
-    // clicked = document.querySelector('.checkbox')
-    // console.log(clicked);
+
 }
 
-// console.log(checkbox, otherBtn, editBtn, deleteBtn)
 
 addBtn.addEventListener('click', init)
 
 
+
+// This checks if an element was added after an event
+displayTodo.addEventListener('DOMNodeInserted', async function(){
+    let listedItems = displayTodo.querySelectorAll('.listed-items')
+    let check = document.querySelectorAll('.listed-items input')
+    let getId = await fetchData()
```

Comment:
If that is the case, this would mean that you are calling this fetchData() function which fetches all the todos every time a new todo is created. This operation is a tad bit expensive.